ush/remap.sh

```diff
@@ -0,0 +1,376 @@
+    varname_array=("${global_array[@]}")    
+    _comma_split_string "${angles}"
+    angle_array=("${global_array[@]}")
+    _strip_whitespace "${varname_array[0]}"
+    xvar="${out_string}"
+    _strip_whitespace "${varname_array[1]}"
+    yvar="${out_string}"
+    nangles="${#angle_array[@]}"
+    
+    echo "Rotating and remapping variables ${xvar} and ${yvar} from file ${varfile}."
+    if [[ "${nangles}" == 1 ]]; then
+	_strip_whitespace "${angle_array[0]}"
+	theta="${out_string}"
+	cdo -expr,"xr=${xvar}*cos(${theta})-${yvar}*sin(${theta}); yr=${xvar}*sin(${theta})+${yvar}*cos(${theta})" -selname,"${xvar}","${yvar}","${theta}" "${varfile}" "${var_rotate_path}"
+    elif [[ "${nangles}" == 2 ]]; then
+	_strip_whitespace "${angle_array[0]}"
+	cosang="${out_string}"
+	_strip_whitespace "${angle_array[1]}"
+	sinang="${out_string}"
+	cdo -expr,"xr=${xvar}*${cosang}-${yvar}*${sinang}; yr=${xvar}*${sinang}+${yvar}*${cosang}" -selname,"${xvar}","${yvar}","${cosang}","${sinang}" "${varfile}" "${var_rotate_path}"
+	
+    else
+	echo "Vector rotations with ${nangles} attributes is not supported. Aborting!!!"
+	exit 102
+    fi
+
+    cdo_remap "xr" "${var_rotate_path}" "${interp_type}"
+    varname_update "xr" "${xvar}" "${output_path}"
+    cdo_remap "yr" "${var_rotate_path}" "${interp_type}"
+    varname_update "yr" "${yvar}" "${output_path}"
+    rm "${var_rotate_path}" >> /dev/null
+}
+
+#######
+
+# varname_update - Rename a variable in a netCDF file and write to a new file.
+#
+# Description:
+#   This function renames a variable in a netCDF file from
+#   `old_varname` to `new_varname` and writes the result to a new
+#   netCDF file specified by `ncfile`.
+#
+# Parameters:
+#   $1 - The old variable name to be renamed.
+#   $2 - The new variable name to use.
+#   $3 - The path to the input netCDF file.
+#
+# Example usage:
+#   varname_update "old_variable" "new_variable" "input_file.nc"
+#
+# This example renames the variable "old_variable" to "new_variable"
+# in the input netCDF file and writes the result to a new netCDF file.
+function varname_update(){
+    local old_varname="${1}"
+    local new_varname="${2}"
+    local ncfile="${3}"
+
+    echo "Renaming variable ${old_varname} to ${new_varname} and writing to file ${ncfile}."
+    ncrename -O -v "${old_varname}","${new_varname}" "${ncfile}"
+}
+
+#######
+
+start_time=$(date +%s)
+_calling_script=$(basename "${BASH_SOURCE[0]}")
+start_time_human=$(date -d"@${start_time}" -u)
+echo "Begin ${_calling_script} at ${start_time_human}."
+
+# Read the configuration file for the the variables to be remapped and
+# proceed accordingly.
+while IFS= read -r line; do
+
+    # Get the attributes for the respective variable(s).
+    varname=$(echo "${line}" | awk '{print $1}')
+    interp_type=$(echo "${line}" | awk '{print $2}')
+    srcgrid=$(echo "${line}" | awk '{print $3}')
+    rotate=$(echo "${line}" | awk '{print $4}')
+    angle=$(echo "${line}" | awk '{print $5}')
```

Comment:
Use here-strings instead of piping an echo, e.g.:
```
    varname=$(awk '{print $1}' <<< "${line}")
```

Reply:
Done.